### ISSUE TITLE
Field reduction transfers

### ISSUE THREAD

**Issue by GiudGiud** [idaholab/moose#23354]
## Reason
With the merge of the general field transfer, we should consider that we could somewhat easily support this.
This would be pretty unique, and a strong advantage for UQ studies. We could run multiphysics with uncertain child apps and transfer the average behavior to the main app for example. 

## Design
Add logic in general field and each derived transfer to send multiple origin values for each target point, and perform a reduction in the general field transfer main class. 

The main pain point will be sending multiple source points for a single target point, in multiple different transfer epochs. This will require some accounting, especially for computing a standard deviation.

## Impact
UQ facilitated

@zachmprince @grmnptr @somu15 any interest on reduction of fields in stochastic tools as opposed to single values?
Or are you using reporters to do this right now?

**Comment by somu15** [comment]
Yes, we are planning to develop field-output Gaussian Processes in the STM. This capability should be useful for FOGP. Right, now, GPs in STM are limited to scalar outputs.

**Comment by somu15** [comment]
From your description, it looks like the field output is reduced to a scalar before sending to the main App. Is this true? Can we send the full field to the main App?

**Comment by GiudGiud** [comment]
Ah which description did I write wrong?
The full fields can be transferred

**Comment by somu15** [comment]
Maybe I mis-understood. Can you give an example of this? "_and perform a reduction in the general field transfer main class_"

**Comment by GiudGiud** [comment]
Subapp 1 sends a field that looks like 'x'
Subapp 2 sends a field that looks like '2x'
General Field main class performs a reduction:
-average, the received field looks like 1.5x
-min, the received field looks like x
and so on

**Comment by somu15** [comment]
Okay, sounds good. As you said, sending the whole field from sub to main would be nice as well.

**Comment by grmnptr** [comment]
I was planning on working on the FieldTransfer with respect to dimensionality reduction. It might not be super obvious since the vectors might be parallelized differently. Would be cool if we could do it without serialization.

**Comment by GiudGiud** [comment]
Depends on the serialization you mean.
Right now we wont need to serialize on the mesh, it can remain distributed.
However for all the app values, this current algorithm will very much need all the child fields to be contributed to the app performing the reduction.

What we could do would be to not store the entire fields then perform the reduction.

**Comment by zachmprince** [comment]
STM has a bunch of [Calculator](https://github.com/idaholab/moose/blob/next/modules/stochastic_tools/include/utils/Calculators.h) objects that can handle the parallelism of sub-apps, the fields will probably need to be serialized though. Right now we have this capability with scalars (PPs), vectors (VPPs), and vector of vectors (Reporters). I don't think it would be a big ask to extend this to field quantities. Maybe we should promote these calculators to the framework and include them in some sort of transfer.

**Comment by GiudGiud** [comment]
I agree for copy-like transfer we could use some similar capability as this. I actually suspected you already had this

But for using more advanced algorithms (shape function evaluation, nearest node etc) I think the Transfers frame is where we want to be

**Comment by somu15** [comment]
> STM has a bunch of [Calculator](https://github.com/idaholab/moose/blob/next/modules/stochastic_tools/include/utils/Calculators.h) objects that can handle the parallelism of sub-apps, the fields will probably need to be serialized though. Right now we have this capability with scalars (PPs), vectors (VPPs), and vector of vectors (Reporters). I don't think it would be a big ask to extend this to field quantities. Maybe we should promote these calculators to the framework and include them in some sort of transfer.

Maybe a naive question: What's the difference between field quantities and vector of vectors?

**Comment by GiudGiud** [comment]
A field is like a variable.
A vector can be a VectorPostprocessor output, but you could totally form a vector (not even a vector of vector) that has all the DoF values of a variable.

**Comment by somu15** [comment]
So, if we want to train a surrogate to learn field quantities, the training output data should be a vector or vectors.

**Comment by GiudGiud** [comment]
Yes you could prefer that format.
You could also choose to interface with fields.
